### ISSUE TITLE
Markers end up tagged as hidden on map 

### ISSUE THREAD

**Issue by ws141** [zcreativelabs/react-simple-maps#73]
With the new ZoomableGlobe feature the markers that are not inbetween of latitude -90 to 90 are being tagged as hidden even if they re not displayed on a globe but rather on a map. This causes them not to react to events such as hover and they end up with different styling (in this case black).

[https://github.com/zcreativelabs/react-simple-maps/blob/master/src/Marker.js#L116](url) is currently the issue and probably needs to be made aware of the context it's in.

Here's the slightly modified simple-markers example showcasing the issue.
<img width="914" alt="screen shot 2018-03-14 at 14 28 57" src="https://user-images.githubusercontent.com/1135081/37405407-d185080a-2794-11e8-902e-0e20b69d7d81.png">

Example code:
```javascript
import React, { Component } from "react"
import {
  ComposableMap,
  ZoomableGroup,
  Geographies,
  Geography,
  Markers,
  Marker,
} from "react-simple-maps"

const wrapperStyles = {
  width: "100%",
  maxWidth: 980,
  margin: "0 auto",
}

const markers = [
  { coordinates: [-90.1, 39.9041] },
  { coordinates: [-40.1, 39.9041] },
  { coordinates: [40.1, 39.9041] },
  { coordinates: [90.1, 39.9041] },
]

class SimpleMarkers extends Component {
  render() {
    return (
      <div style={wrapperStyles}>
        <ComposableMap
          projectionConfig={{ scale: 200 }}
          width={1000}
          height={500}
          style={{
            width: "100%",
            height: "auto",
          }}
          >
          <ZoomableGroup center={[ 0, 0 ]} disablePanning>
            <Geographies geography="/static/world-50m.json">
              {(geographies, projection) =>
                geographies.map((geography, i) => (
                    <Geography
                      key={i}
                      geography={geography}
                      projection={projection}
                      style={{
                        default: {
                          fill: "#ECEFF1",
                          stroke: "#607D8B",
                          strokeWidth: 0.75,
                          outline: "none",
                        },
                        hover: {
                          fill: "#CFD8DC",
                          stroke: "#607D8B",
                          strokeWidth: 0.75,
                          outline: "none",
                        },
                        pressed: {
                          fill: "#FF5722",
                          stroke: "#607D8B",
                          strokeWidth: 0.75,
                          outline: "none",
                        },
                      }}
                    />
                  )
                )
              }
            </Geographies>
            <Markers>
              {markers.map((marker, i) => (
                <Marker
                  key={i}
                  marker={marker}
                  style={{
                    default: { fill: "#FF5722" },
                    hover: { fill: "#FFFFFF" },
                    pressed: { fill: "#FF5722" },
                  }}
                  >
                  <circle
                    cx={0}
                    cy={0}
                    r={10}
                    style={{
                      stroke: "#FF5722",
                      strokeWidth: 3,
                      opacity: 0.9,
                    }}
                  />
                </Marker>
              ))}
            </Markers>
          </ZoomableGroup>
        </ComposableMap>
      </div>
    )
  }
}

export default SimpleMarkers
```

**Comment by zimrick** [comment]
Thanks for flagging @ws141 will have a look at this.

**Comment by ws141** [comment]
Np @zimrick I could attempt to write a fix for it if you have any suggestion for how you'd like such a thing to be implemented.

I was thinking maybe through a property that is set by default to false that would trigger the line 116 in Marker.js to run if set to true. Have you got any better suggestion?

**Comment by zimrick** [comment]
Yup, it would definitely go in this direction. What I am wondering about is what to check for. We could use a `globe` property, like the one used on `<Graticule />`. However, I wonder if it would not be better to check something like `projection.clipAngle === 90` instead so that users don't have to add the `globe` prop to markers every time. A manual override should probably still be possible, but it should have some good preset behaviour.

**Comment by ws141** [comment]
@zimrick thanks for the tip on `clipAngle()`.  

I decided to have a go at a fix for this but steered away from having the user override since it would allow for some weird behavior if markers would be able to be forced to become hidden on the map and end up looking like the bug I described. 
It would also allow for markers never to be hidden on a globe which would weird as well.
Lastly just having the override only be available in the case of the marker being placed on a map it would cause the property globe to only work in certain situations and seem buggy in the globe context.

Looking forward to your feedback.
